fenix/app/src/androidTest/java/org/mozilla/fenix/experimentintegration/SurveyExperimentIntegrationTest.kt

```diff
@@ -0,0 +1,61 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package org.mozilla.fenix.experimentintegration
+
+import android.widget.TextView
+import androidx.test.platform.app.InstrumentationRegistry
+import androidx.test.uiautomator.UiDevice
+import androidx.test.uiautomator.UiSelector
+import okhttp3.mockwebserver.MockWebServer
+import org.junit.After
+import org.junit.Before
+import org.junit.Rule
+import org.junit.Test
+import org.mozilla.fenix.ext.settings
+import org.mozilla.fenix.helpers.AndroidAssetDispatcher
+import org.mozilla.fenix.helpers.HomeActivityTestRule
+import org.mozilla.fenix.helpers.TestHelper
+import org.mozilla.fenix.ui.robots.*
+
+
+class SurveyExperimentIntegrationTest {
+    private lateinit var mDevice: UiDevice
+    private lateinit var mockWebServer: MockWebServer
+
+    @get:Rule
+    val activityTestRule = HomeActivityTestRule()
+
+    @Before
+    fun setUp() {
+        mDevice = UiDevice.getInstance(InstrumentationRegistry.getInstrumentation())
+        mockWebServer = MockWebServer().apply {
+            dispatcher = AndroidAssetDispatcher()
+            start()
+        }
+        TestHelper.appContext.settings().showSecretDebugMenuThisSession = true
+    }
+
+    @After
+    fun tearDown() {
+        homeScreen {
+        }.openThreeDotMenu {
+        }.openSettings {
+        }.openExperimentsMenu{
+            getExperiment("Viewpoint")
+        }
+        mockWebServer.shutdown()
+    }
+
+    @Test
+    fun checkSurveyNavigatesCorrectly() {
+        mDevice.findObject(
+            UiSelector().className(
+                TextView::class.java
+            ).textContains("Take Survey")).click()
```

Comment:
I think it's fine to drop the `className(TextView::class.java)` bit and just call `textContains`. Alternatively, preferably, if it has a `resourceId`, you can call `.resourceId("$packageName:id/insertidhere")` or even by string resource `mDevice.findObject(UiSelector().text(getStringResource(R.takesurveystring)))`

Reply:
I wish I knew how to get the actual names. I don't know the variable name that the string uses, or how to find it really. I could use some help on that

Reply:
No problem. Pontoon lets us look-up that data [1] for Fenix. We can see a preference `preferences_take_survey` string, thus we can access the string directly with:

`mDevice.findObject(UiSelector().text(getStringResource(R.string. preferences_take_survey))`

[1] https://pontoon.mozilla.org/en-CA/firefox-for-android/mozilla-mobile/fenix/app/src/main/res/values/strings.xml/?search=Survey&string=276852